maskrcnn_benchmark/modeling/matcher.py

```diff
@@ -51,9 +51,15 @@ def __call__(self, match_quality_matrix):
             be matched.
         """
         if match_quality_matrix.numel() == 0:
-            # handle empty case
-            device = match_quality_matrix.device
-            return torch.empty((0,), dtype=torch.int64, device=device)
+            # empty targets or proposals not supported during training
+            if match_quality_matrix.shape[0] == 0:
+                raise ValueError(
+                    "No ground-truth boxes available for one of the images "
```

Comment:
make it tell the user which image / filename if possible. otherwise it becomes a bit of a nightmare to track back

Reply:
I agree as well, but for the image names to be displayed in the error message means that the assert would need to be applied in the user code as well, when they create a new Dataset class